app/jobs/calendar_importer/events/linked_data_event.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module CalendarImporter::Events
+  class LinkedDataEvent < Base
+    attr_reader :uid, :start_time, :end_time, :summary, :description, :location
+
+    def read_value_of(data, field_name)
+      field_data = data[field_name]
+      return unless field_data
+
+      field_data['@value']
+    end
+
+    def parse_timestamp(value)
+      DateTime.parse value.to_s
+    rescue Date::Error
+      nil
+    end
+
+    def extract_location(data)
+      loc = data['location']
+      return if loc.blank?
+
+      address = loc['address']
+      address = address['street_address'] if address.is_a?(Hash)
+      address
+    end
+
+    def initialize(data)
+      super data
+
+      @url = data['url']
+      return if @url.blank?
+
+      @description = data['description']
+      # return if @description.blank?
+
+      @start_time = parse_timestamp(read_value_of(data, 'start_date'))
+      return if @start_time.blank?
+
+      @end_time = parse_timestamp(read_value_of(data, 'end_date'))
+      return if @end_time.blank?
+
+      @summary = data['name']
+      return if @summary.blank?
+
+      @location = extract_location(data)
+      # return if @location.blank?
+
+      @uid = @url
+      @is_valid = true
```

Comment:
Instead of exiting with partial data, wouldn't this be better to set `@is_valid` to check for presence of the required fields? So we always populate the fields we can find even if it's parked as invalid at the end?